cpp/src/groupby/sort/group_rank_scan.cu

```diff
@@ -115,13 +118,47 @@ std::unique_ptr<column> dense_rank_scan(column_view const& order_by,
     order_by,
     group_labels,
     group_offsets,
-    [] __device__(bool equality, auto row_index) { return equality; },
+    [] __device__(bool unequal, auto row_index_in_group) { return unequal ? 1 : 0; },
```

Comment:
Boolean is `0` or `1` when casting to integer.
```suggestion
    [] __device__(bool unequal, size_type) { return unequal; },
```

Reply:
If you prefer clarity then we can just have:
```suggestion
[] __device__(bool unequal, size_type) { return unequal ? 1 : 0; },
```

Reply:
Fixed.